models/search_args.go

```diff
@@ -1,12 +1,21 @@
 package models
 
+import (
+	"slices"
+
+	"github.com/samber/lo"
+)
+
+var CollapseFromFacetLine = 2
+
 type SearchArgs struct {
-	Query    string              `query:"q,omitempty"`
-	Filters  map[string][]string `query:"f,omitempty"`
-	Page     int                 `query:"page"`
-	Sort     []string            `query:"sort,omitempty"`
-	PageSize int                 `query:"page-size"`
-	Facets   []string            `query:"-"`
+	Query      string              `query:"q,omitempty"`
+	Filters    map[string][]string `query:"f,omitempty"`
+	Page       int                 `query:"page"`
+	Sort       []string            `query:"sort,omitempty"`
+	PageSize   int                 `query:"page-size"`
+	Facets     []string            `query:"-"`
+	FacetLines [][]string          `query:"-"`
```

Comment:
Not sure if this `query:"-"` annotation at the end is right here (or what it does).